src/meta-srv/src/election/rds/mysql.rs

```diff
@@ -203,107 +165,352 @@ impl Executor<'_> {
+    }
+
+    fn set_session_lock_wait_timeout_sql() -> &'static str {
+        "SET SESSION innodb_lock_wait_timeout = ?;"
+    }
+
+    fn set_session_max_execution_time_sql() -> &'static str {
+        "SET SESSION max_execution_time = ?;"
+    }
+
+    fn create_table_sql(&self) -> String {
+        format!(
+            r#"
+            CREATE TABLE IF NOT EXISTS `{}` (
+                k VARBINARY(3072) PRIMARY KEY,
+                v BLOB
+            );"#,
+            self.table_for_election
+        )
+    }
+
+    fn insert_once_sql(&self) -> String {
+        format!(
+            "INSERT IGNORE INTO `{}` (k, v) VALUES ('__place_holder_for_lock', '');",
+            self.table_for_election
+        )
+    }
+
+    fn check_version_sql(&self) -> String {
+        "SELECT @@version;".to_string()
+    }
+
+    async fn reset_client(&mut self) -> Result<()> {
+        self.current = None;
+        self.maybe_init_client().await
+    }
+
+    async fn ensure_table_exists(&mut self) -> Result<()> {
+        let create_table_sql = self.create_table_sql();
+        let query = sqlx::query(&create_table_sql);
+        self.execute(query, &create_table_sql).await?;
+        // Insert at least one row for `SELECT * FOR UPDATE` to work.
+        let insert_once_sql = self.insert_once_sql();
+        let query = sqlx::query(&insert_once_sql);
+        self.execute(query, &insert_once_sql).await?;
+        Ok(())
+    }
+
+    async fn maybe_init_client(&mut self) -> Result<()> {
+        if self.current.is_none() {
+            let client = self.pool.acquire().await.context(AcquireMySqlClientSnafu)?;
+
+            self.current = Some(client);
+            if !self.wait_timeout.is_zero() {
+                // Set wait timeout to avoid idle connection.
+                let set_session_wait_timeout_sql = Self::set_session_wait_timeout_sql();
+                let query =
+                    sqlx::query(set_session_wait_timeout_sql).bind(self.wait_timeout.as_secs());
+                self.execute(query, set_session_wait_timeout_sql).await?;
+            }
+            // Set lock wait timeout to avoid waiting too long.
+            let set_session_lock_wait_timeout_sql = Self::set_session_lock_wait_timeout_sql();
+            let query = sqlx::query(set_session_lock_wait_timeout_sql)
+                .bind(self.innode_lock_wait_timeout.as_secs());
+            self.execute(query, set_session_lock_wait_timeout_sql)
+                .await?;
+            // Set max execution time to avoid long-running queries.
+            let set_session_max_execution_time_sql = Self::set_session_max_execution_time_sql();
+            let query = sqlx::query(set_session_max_execution_time_sql)
+                .bind(self.max_execution_time.as_millis() as u64);
+            self.execute(query, set_session_max_execution_time_sql)
+                .await?;
+            // Set session isolation level.
+            self.set_session_isolation_level().await?;
```

Comment:
Can all these operations be executed in one transaction?

Reply:
resolved in [bf65ffc](https://github.com/GreptimeTeam/greptimedb/pull/6341/commits/bf65ffc23bb5977e295c36755e4e07a4bf07417a)